addon/errors.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Checks if the given response is a FetchError
+ */
+export function isFetchError(response: Response): boolean {
+  return response.ok === false
+}
+
+/**
+ * Checks if the given response represents an unauthorized request error
+ */
+export function isUnauthorizedError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 401;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a forbidden request error
+ */
+export function isForbiddenError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 403;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents an invalid request error
+ */
+export function isInvalidError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 422;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a bad request error
+ */
+export function isBadRequestError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 400;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a "not found" error
+ */
+export function isNotFoundError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 404;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a "gone" error
+ */
+export function isGoneError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 410;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents an "abort" error
+ */
+export function isAbortError(response: Response): boolean {
```

Comment:
`AbortError` is from rejecting the `fetch` promise rather than setting status code to 0.
https://dom.spec.whatwg.org/#aborting-ongoing-activities

You may want to make a convenient method for `err.name == 'AbortError'`.